### ISSUE TITLE
Image Compare: control has no label for screen readers (A11y)

### ISSUE THREAD

**Issue by ActiveB1t** [Automattic/jetpack#17278]
#### Is your feature request related to a problem? Please describe.
On keyboard navigation and while using a screen reader, the control for the Image Compare block has no label.

#### Describe the solution you'd like
An addition of a label, through an aria-label attribute or otherwise, that communicates the function of the element to the screen reader user.
